### ISSUE TITLE
Problem with cycle function

### ISSUE THREAD

**Issue by Jyothsnasrinivas** [rahulmutt/ghcvm#67]
```haskell

module Main where

main = do
  print $ take 10 (cycle [1,2,3])

```
It works for take 1, 2 & 3, beyond that it hangs. It seems to be a problem with the cycle function.

**Comment by Jyothsnasrinivas** [comment]
```haskell
module Main where

main = do
  print $ take 10 (repeat 5)
```
Seems to be a problem with infinite data structures.

**Comment by rahulmutt** [comment]
Cycle fixed in #70, repeat has a new issue reported in #71
